### PR TITLE
Registry index management service

### PR DESCRIPTION
**Pull request by kithminisg**
* Acquire the states related to the Registry (Hook Acc.).
* Update the Firestore after filtering the corresponding states for the transaction.
* Bulk update feature. (recovery purpose).

### CODE REVIEW

File: registry-index-service/index-manager.js [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+                stateKeyTokenId = await this.#generateTokenIdStateKey(nftTokenId);
+                affectedStates.push(stateKeyTokenId);
+                break;
+        }
+
+        await this.#persisit(affectedStates);
+        console.log(`|${trx.Account}|${memoType}|Completed Transaction Handle`);
+    }
+
+    async #generateTokenIdStateKey(nfTokenId) {
```
Comment (chalith) [inline registry-index-service/index-manager.js:231]: Shall we move this to state helper in js lib

File: registry-index-service/index-manager.js [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+                delete indexUpdates.set.configs[decoded.key];
+                indexUpdates.delete.configs[decoded.key] = true;
+            }
+        }
+
+        const persistIndex = async () => {
+            // Persist the data to the firestore index.
+            // Since we only have one list object per entry,
+            // There's only one insert, update or delete operation per document.
+            // So we can promise.all all the inserts,updates and deletes.
```
Comment (chalith) [inline registry-index-service/index-manager.js:307]: all

File: registry-index-service/package.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        "build": "npm run lint && ncc build index-manager.js --minify -o dist"
+    },
+    "keywords": [],
+    "author": "",
+    "license": "ISC",
+    "dependencies": {
+        "evernode-js-client": "0.4.38",
+        "jsonwebtoken": "8.5.1",
+        "fs": "0.0.1-security",
+        "path": "^0.12.7",
```
Comment (chalith) [inline registry-index-service/package.json:18]: Use 0.12.7 (remove ^)

File: registry-index-service/package.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        "util": "^0.12.4"
```
Comment (chalith) [inline registry-index-service/package.json:19]: Same

File: registry-index-service/index-manager.js [hunk 5]
```diff
@@ -0,0 +1,374 @@
+const fs = require('fs');
+const process = require('process');
+const path = require('path');
+const codec = require('ripple-address-codec');
+const evernode = require('evernode-js-client');
+const { XrplApi, StateHelpers, RegistryClient, Defaults, RegistryEvents, } = require('evernode-js-client');
```
Comment (SavindaSenevirathne) [inline registry-index-service/index-manager.js:6]: Same package import two times. I feel we can avoid the 5th line and add those components to the 6th line.

File: registry-index-service/README.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+### Generating the firebase key file for state index
+- Go to [firebase console](https://console.firebase.google.com).
+- Choose your project.
+- Go to `Project Settings`.
+- Go to `Service Accounts` tab.
+- Download a key json file with `Generate new private key` button.
+- Place it under `service-acc` directory as `./registry-index-service/service-acc/firebase-sa-key.json`.
+- If your system is 64-bit, You need to have 32-bit support in you system to run emulator wrappers.
+- 32-bit support can be installed with `apt-get install libc6-i386`.
```
Comment (chalith) [inline registry-index-service/README.md:12]: We don't need this right

**Review by SavindaSenevirathne — APPROVED** [review]

**Pull request merged**